File: app/src/main/java/com/yawar/memo/modelView/ForwardDialogViewModel.java
package com.yawar.memo.modelView;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.yawar.memo.model.ChatRoomModel;
import com.yawar.memo.repositry.ChatMessageRepoo;
import com.yawar.memo.repositry.ChatRoomRepoo;
import com.yawar.memo.utils.BaseApp;

import java.util.ArrayList;

public class ForwardDialogViewModel extends ViewModel {
    BaseApp baseApp = BaseApp.getInstance();
    private final ChatRoomRepoo repository = baseApp.getChatRoomRepoo();
    private final ChatMessageRepoo chatMessageRepo = baseApp.getChatMessageRepoo();

    public MutableLiveData<ArrayList<ChatRoomModel>> chatRoomListMutableLiveData;
    public final ArrayList<ChatRoomModel> chatRoomsList = new ArrayList<>();

    public LiveData<ArrayList<ChatRoomModel>> loadData() {
        return repository.getChatRoomListMutableLiveData();

    }

    public  void clearSelectedMessage() {
         chatMessageRepo.clearSelectedMessage();

    }

}

File: app/src/main/java/com/yawar/memo/ui/forwardPage/ForwardDialogViewModel.kt
package com.yawar.memo.ui.forwardPage

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.yawar.memo.model.ChatRoomModel
import com.yawar.memo.repositry.ChatMessageRepoo
import com.yawar.memo.repositry.ChatRoomRepoo
//import com.yawar.memo.repositry.chatRoomRepo.ChatRoomRepoImp
import com.yawar.memo.utils.BaseApp
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

//@HiltViewModel
//class ForwardDialogViewModel @Inject constructor(private val repository: ChatRoomRepoImp) : ViewModel() {
@HiltViewModel
class ForwardDialogViewModel@Inject constructor(private val chatMessageRepo:ChatMessageRepoo, private val repository: ChatRoomRepoo) : ViewModel() {

    var baseApp = BaseApp.getInstance()
//    private val repository = baseApp.chatRoomRepoo
//    private val chatMessageRepo = baseApp.chatMessageRepoo


    fun loadData(): LiveData<ArrayList<ChatRoomModel?>?> {
        return repository.chatRoomListMutableLiveData as LiveData<ArrayList<ChatRoomModel?>?>
    }
    fun clearSelectedMessage() {
        chatMessageRepo.clearSelectedMessage()
    }

}